### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by molly-moen**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
Update README to be in line with new template and dotnet version

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,89 +2,87 @@
...
-1. Gets a list of alarms for specific parameters
-1. Gets a single alarm
-1. Modifies alarm status
-1. Create/Read/Update/Delete Rules
-    1. Create Rules
-    1. Gets a list of rules for specific parameters
-    1. Gets a single rule
-    1. Modify existing rule
-    1. Delete existing rule
+This microservice was built as part of the [Azure IoT Remote Monitoring](https://github.com/Azure/azure-iot-pcs-remote-monitoring-java) project to provide a generic implementation for an end-to-end IoT solution. More information [here][rm-arch-url].
```
Comment (dluc) [inline README.md]: please split long lines, wrap at 76-79th char

File: README.md [hunk 2]
```diff
@@ -157,30 +151,55 @@ dockerBuildOptions ++= Seq("--compress", "--label", "Tags=azure,iot,pcs,telemetr
...
+  settings will persist across terminal sessions and reboots.
+* For Linux and OSX environments, the [env-vars-setup](scripts/env-vars-setup)
+  script needs to be executed every time a new console is opened.
+  Depending on the OS and terminal, there are ways to persist values
+  globally, for more information these pages should help:
+  * https://stackoverflow.com/questions/13046624/how-to-permanently-export-a-variable-in-linux
+  * https://stackoverflow.com/questions/135688/setting-environment-variables-in-os-x
+  * https://help.ubuntu.com/community/EnvironmentVariables
+* IntelliJ IDEA: env. vars can be set in each Run Configuration, see
+  https://www.jetbrains.com/help/idea/run-debug-configuration-application.html
```
Comment (dluc) [inline README.md]: please make the link clickable

File: README.md [hunk 3]
```diff
@@ -157,30 +151,55 @@ dockerBuildOptions ++= Seq("--compress", "--label", "Tags=azure,iot,pcs,telemetr
...
-Please follow our contribution guildelines and code style conventions defined in [guidelines](https://github.com/Azure/iothub-manager-java/blob/master/CONTRIBUTING.md).
+* For Windows, the variables can be set [in the system][windows-envvars-howto-url]
+  as a one time only task. The [env-vars-setup.cmd](scripts/env-vars-setup.cmd)
+  script needs to be prepared and executed just once. When executed, the
+  settings will persist across terminal sessions and reboots.
+* For Linux and OSX environments, the [env-vars-setup](scripts/env-vars-setup)
+  script needs to be executed every time a new console is opened.
+  Depending on the OS and terminal, there are ways to persist values
+  globally, for more information these pages should help:
+  * https://stackoverflow.com/questions/13046624/how-to-permanently-export-a-variable-in-linux
```
Comment (dluc) [inline README.md]: make links clickable

File: README.md [hunk 4]
```diff
@@ -157,30 +151,55 @@ dockerBuildOptions ++= Seq("--compress", "--label", "Tags=azure,iot,pcs,telemetr
...
+The configuration file in the repository references some environment
+variables that need to created at least once. Depending on your OS and
+the IDE, there are several ways to manage environment variables:
 
-Please follow our contribution guildelines and code style conventions defined in [guidelines](https://github.com/Azure/iothub-manager-java/blob/master/CONTRIBUTING.md).
+* For Windows, the variables can be set [in the system][windows-envvars-howto-url]
+  as a one time only task. The [env-vars-setup.cmd](scripts/env-vars-setup.cmd)
+  script needs to be prepared and executed just once. When executed, the
+  settings will persist across terminal sessions and reboots.
+* For Linux and OSX environments, the [env-vars-setup](scripts/env-vars-setup)
```
Comment (dluc) [inline README.md]: nit: replace "OSX" with "MacOS"

File: README.md [hunk 5]
```diff
@@ -157,30 +151,55 @@ dockerBuildOptions ++= Seq("--compress", "--label", "Tags=azure,iot,pcs,telemetr
 dockerEntrypoint := Seq("bin/telemetry")
 ```
 
-The package logic is executed via
-[sbt-native-packager](https://github.com/sbt/sbt-native-packager), installed
-in [plugins.sbt](project/plugins.sbt).
+The package logic is executed via [sbt-native-packager](https://github.com/sbt/sbt-native-packager), installed in [plugins.sbt](project/plugins.sbt).
```
Comment (dluc) [inline README.md]: split long line

File: README.md [hunk 6]
```diff
@@ -2,89 +2,87 @@
...
-Device Telemetry Overview
-=========================
+# Device Telemetry Overview
 
-This service offers read access to device telemetry, full CRUD for rules, and read/write for
- alarms from storage for the client via a RESTful endpoint.
+This service provides a RESTful endpoint for read access to device telemetry, full CRUD for rules, and read/write for alarms from storage.
 
-## Features the microservice offers:
+## Why?
```
Comment (dluc) [inline README.md]: nit: maybe "Purpose" sounds better than "Why?"

File: README.md [hunk 7]
```diff
@@ -2,89 +2,87 @@
...
-    1. Gets a list of rules for specific parameters
-    1. Gets a single rule
-    1. Modify existing rule
-    1. Delete existing rule
+This microservice was built as part of the [Azure IoT Remote Monitoring](https://github.com/Azure/azure-iot-pcs-remote-monitoring-java) project to provide a generic implementation for an end-to-end IoT solution. More information [here][rm-arch-url].
 
-Dependencies
-============
+## Features
+* Gets a list of telemetry messages for specific parameters
```
Comment (dluc) [inline README.md]: "Gets a list of telemetry messages"

File: README.md [hunk 8]
```diff
@@ -2,89 +2,87 @@
...
+* Gets a list of alarms for specific parameters
```
Comment (dluc) [inline README.md]: "Gets a list of alarms"

File: README.md [hunk 9]
```diff
@@ -2,89 +2,87 @@
...
+* Gets a single alarm
+* Modifies alarm status
+* Create/Read/Update/Delete Rules
+    * Create Rules
+    * Gets a list of rules for specific parameters
```
Comment (dluc) [inline README.md]: "Gets a list of rules"

File: README.md [hunk 10]
```diff
@@ -2,89 +2,87 @@
...
+This microservice was built as part of the [Azure IoT Remote Monitoring](https://github.com/Azure/azure-iot-pcs-remote-monitoring-java) project to provide a generic implementation for an end-to-end IoT solution. More information [here][rm-arch-url].
 
-Dependencies
-============
+## Features
+* Gets a list of telemetry messages for specific parameters
+* Gets a list of alarms for specific parameters
+* Gets a single alarm
+* Modifies alarm status
+* Create/Read/Update/Delete Rules
```
Comment (dluc) [inline README.md]: I think this is sufficient, no need for the following lines

File: README.md [hunk 11]
```diff
@@ -2,89 +2,87 @@
...
 
-Run `scripts\env-vars-setup.cmd` on Windows or `source scripts\env-vars-setup`
-on Mac/Linux to set up the environment variables needed to run the service locally.
-If using envornemnt variables, this service requires the following environment
-variables to be set:
-- PCS_TELEMETRY_DOCUMENTDB_CONNSTRING - the connection string for the Azure DocumentDB backend
-- PCS_STORAGEADAPTER_WEBSERVICE_URL - the url for
-  the [Storage Adapter Webservice](https://github.com/Azure/pcs-storage-adapter-dotnet)
-  used for key value storage
+You can run the microservice and its dependencies using [Docker](https://www.docker.com/) with the instructions [here][run-with-docker-url].
```
Comment (dluc) [inline README.md]: split long line

File: README.md [hunk 12]
```diff
@@ -2,89 +2,87 @@
...
 
-## Quickstart - Running the service with Docker
+## Running the service locally
 
---todo
+## Prerequisites
+### 1. Deploy Azure Services
+This service has a dependency on the following Azure resources. Follow the instructions for [Deploy the Azure services](https://docs.microsoft.com/azure/iot-suite/iot-suite-remote-monitoring-deploy-local#deploy-the-azure-services).
```
Comment (dluc) [inline README.md]: long line

File: README.md [hunk 13]
```diff
@@ -2,89 +2,87 @@
...
+### 2. Setup Dependencies
 
-### Prerequisites
+This service depends on the following repository.
+1. [Storage Adapter Microservice](https://github.com/Azure/pcs-storage-adapter-java)
 
-- Install SBT: http://www.scala-sbt.org/download.html
-- Install the latest Java SDK: http://www.oracle.com/technetwork/java/javase/downloads/index.html
+### 3. Environment variables required to run the service
+In order to run the service, some environment variables need to be created at least once. See specific instructions for IDE or command line setup below for more information. More information on environment variables [here](#configuration-and-environment-variables).
```
Comment (dluc) [inline README.md]: long line

File: README.md [hunk 14]
```diff
@@ -2,89 +2,87 @@
...
+  * `PCS_TELEMETRY_DOCUMENTDB_CONNSTRING` = {your Azure Cosmos DB connection string}
+  * `PCS_STORAGEADAPTER_WEBSERVICE_URL` = http://localhost:9022/v1
 
-## Run and Debug with IntelliJ
+# Running the service in an IDE
 
-Install Intellij IDEA Community: https://www.jetbrains.com/idea/download
+## Prerequisites
+- Install Intellij IDEA Community: https://www.jetbrains.com/idea/download
```
Comment (dluc) [inline README.md]: typo: "IntelliJ"

Comment (dluc) [reply README.md]: you can also remove "Community", because either Ultimate or Community editions are fine

File: README.md [hunk 15]
```diff
@@ -2,89 +2,87 @@
...
+- Install SBT: http://www.scala-sbt.org/download.html
+- Install the latest Java 8 SDK: 
+  http://www.oracle.com/technetwork/java/javase/downloads/index.html
```
Comment (dluc) [inline README.md]: nit: change link to http://www.oracle.com/technetwork/java/javase/downloads

File: README.md [hunk 16]
```diff
@@ -2,89 +2,87 @@
...
 
-Intellij IDEA lets you quickly open the application without using a command
+## Running the service with IntelliJ IDEA
+Intellij IDEA lets you open the application without using a command
```
Comment (dluc) [inline README.md]: typo: IntelliJ

File: README.md [hunk 17]
```diff
@@ -2,89 +2,87 @@
...
-
-* "Open" the project with IntelliJ, the IDE should automatically recognize the
-  SBT structure. Wait for the IDE to download some dependencies (see IntelliJ
-  status bar)
-* Create a new Run Configuration, of type "SBT Task" and enter "run 9004"
-  (including the double quotes). This ensures that the service to start using
-  the TCP port 9004
+building the project (more info [here](https://www.playframework.com/documentation/2.6.x/IDE)).
+
+Steps using IntelliJ IDEA Community, with SBT plugin enabled:
```
Comment (dluc) [inline README.md]: nit: remove "Community"

File: README.md [hunk 18]
```diff
@@ -2,89 +2,87 @@
...
+
+* Make sure the [prerequisites](#prerequisites) are set up.
+* "Open" the project with IntelliJ, the IDE should automatically recognize
+  the SBT structure. Wait for the IDE to download some dependencies
+  (see IntelliJ status bar). This may take a while, hang in there!
+* Create a new Run Configuration, of type "SBT Task", with any name.
+  * Enter "run 9004" (including the double quotes) in Tasks. This ensures that
```
Comment (dluc) [inline README.md]: nit: use code quote:

> Enter `"run 9004"`

File: README.md [hunk 19]
```diff
@@ -2,89 +2,87 @@
...
-And as a nice extra: if you edit the code, you don't need to stop/build/restart
-the application. Play and SBT automatically recompile the files modified on the
-fly. You can also see the re-build log in the Run Tool Window.
-
-## Run and Debug with Eclipse
-
-The integration with Eclipse requires the
-[sbteclipse plugin](https://github.com/typesafehub/sbteclipse), already
+## Running the service with Eclipse
+The integration with Eclipse requires the [sbteclipse plugin](https://github.com/typesafehub/sbteclipse), already
```
Comment (dluc) [inline README.md]: long line

File: README.md [hunk 20]
```diff
@@ -103,41 +101,37 @@ Steps using Eclipse Oxygen ("Eclipse for Java Developers" package):
 * After saving this configuration, you can click "Debug" to connect to the
   running application.
 
-Project Structure
-=================
+# Running the service from the command line
+
+1. Make sure the [prerequisites](#prerequisites) are set up.
+1. Set the following environment variables in your system. More information on environment variables [here](#configuration-and-environment-variables).
```
Comment (dluc) [inline README.md]: long line

File: README.md (hunk 2 above)
Comment (molly-moen) [reply README.md]: link is clickable for me, are there browsers/views where this isn't clickable?

File: README.md (hunk 3 above)
Comment (molly-moen) [reply README.md]: links are clickable

File: README.md (hunk 6 above)
Comment (molly-moen) [reply README.md]: we put Why? in the template, so a few others already have Why?

**Pull request merged**